MobileWallet/Screens/Home/TransactionHistory/TransactionsTableViewController.swift

```diff
@@ -127,12 +135,92 @@ class TransactionsTableViewController: UITableViewController {
         view.backgroundColor = Theme.shared.colors.transactionTableBackground
     }
 
-    @objc private func refreshPullTransactions(_ sender: UIRefreshControl) {
-        refreshTransactionControl.attributedTitle = NSAttributedString(string: "Resfreshing...") //TODO local
-        DispatchQueue.main.asyncAfter(deadline: .now() + 2, execute: {
-            self.tableView.reloadData()
-            sender.endRefreshing()
+    private func showRefreshingSpinner(_ show: Bool) {
+        refreshTransactionControl.attributedTitle = nil
+
+        if show {
+            refreshTransactionControl.tintColor = .systemGray
+            refreshTransactionControl.subviews.first?.alpha = 1
+        } else {
+            UIView.animate(withDuration: 0.25, animations: { [weak self] in
+                guard let self = self else { return }
+                self.refreshTransactionControl.tintColor = .clear
+                self.refreshTransactionControl.subviews.first?.alpha = 0
+            }) { _ in
+                //Switch off the pull to refresh part
+                self.refreshTransactionControl.endRefreshing()
+            }
+        }
+    }
+
+    private func listenForRefreshUpdates() {
+        self.animatedRefresher.setupView(.loading)
+
+        DispatchQueue.main.asyncAfter(deadline: .now() + 0.5, execute: { [weak self] in
+            guard let self = self else { return }
+            //We know where refreshing if this is the first load
+            self.beginRefreshing()
         })
+
+        //After first load if they bring the app into the foreground listen for the connection starting
+        TariEventBus.onMainThread(self, eventType: .torPortsOpened) { [weak self] (_) in
+            guard let self = self else { return }
+            self.beginRefreshing()
+        }
+
+        //Listen for when to stop refreshing
+        TariEventBus.onMainThread(self, eventType: .baseNodeSyncComplete) { [weak self] (result) in
+            guard let self = self else { return }
+
+            if let success: Bool = result?.object as? Bool {
+                if success == true {
+                    self.endRefreshingWithSuccess()
+                } else {
+                    //TODO base node always fails the first few times. When that stabalises we can show an error here.
```

Comment:
Maybe we can agree on a number of max base-node sync retries before failing - I implemented a similar thing on Android for the first step of the new 3-step sending progress.

Reply:
True. Was going to wait until Simian does his multi base node support to see how this changes first